test/extended/baremetal/hosts.go

```diff
@@ -88,9 +140,48 @@ var _ = g.Describe("[sig-installer][Feature:baremetal] Baremetal platform should
 		o.Expect(hosts.Items).ToNot(o.BeEmpty())
 
 		for _, h := range hosts.Items {
-			expectStringField(h, "status.operationalStatus").To(o.BeEquivalentTo("OK"))
-			expectStringField(h, "status.provisioning.state").To(o.Or(o.BeEquivalentTo("provisioned"), o.BeEquivalentTo("externally provisioned")))
-			expectBoolField(h, "spec.online").To(o.BeTrue())
+			expectStringField(h, "baremetalhost", "status.operationalStatus").To(o.BeEquivalentTo("OK"))
+			expectStringField(h, "baremetalhost", "status.provisioning.state").To(o.Or(o.BeEquivalentTo("provisioned"), o.BeEquivalentTo("externally provisioned")))
+			expectBoolField(h, "baremetalhost", "spec.online").To(o.BeTrue())
+		}
+	})
+
+	g.It("have hostfirmwaresetting resources", func() {
+		skipIfNotBaremetal(oc)
+
+		dc := oc.AdminDynamicClient()
+
+		bmc := baremetalClient(dc)
+		hosts, err := bmc.List(context.Background(), v1.ListOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(hosts.Items).ToNot(o.BeEmpty())
+
+		hfsClient := hostfirmwaresettingsClient(dc)
+
+		for _, h := range hosts.Items {
+			hostName := getStringField(h, "baremetalhost", "metadata.name")
+
+			g.By("check that each baremetalhost has a corresponding hostfirmwaresettings")
+			hfs, err := hfsClient.Get(context.Background(), hostName, v1.GetOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(hfs).NotTo(o.Equal(nil))
+
+			// Reenable this when fix to prevent settings with 0 entries is in BMO
+			// g.By("check that hostfirmwaresettings settings have been populated")
+			// expectStringMapField(*hfs, "hostfirmwaresettings", "status.settings").ToNot(o.BeEmpty())
+
+			g.By("check that hostfirmwaresettings conditions show resource is valid")
+			expectSliceField(*hfs, "hostfirmwaresettings", "status.conditions").ToNot(o.BeEmpty())
```

Comment:
Isn't something already somehow addressed within the `checkConditionStatus` method?

Reply:
I think its necessary to check that conditions is not empty but I've moved this check to `checkConditionStatus` and simplified it.